x-pack/beatless/provider/aws/aws.go

```diff
@@ -0,0 +1,41 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package aws
+
+import (
+	"github.com/elastic/beats/libbeat/feature"
+	"github.com/elastic/beats/x-pack/beatless/provider"
+)
+
+// Bundle exposes the trigger supported by the AWS provider.
+var Bundle = provider.MustCreate(
+	"aws",
+	provider.NewDefaultProvider("aws", NewCLI),
+	feature.NewDetails("AWS Lambda", "listen to events on AWS lambda", feature.Experimental),
+).MustAddFunction("cloudwatch_logs",
```

Comment:
I've wrapped the feature API inside a builder pattern, because the way we have designed the registry is to have a flat view of the namespace keys, using a builder pattern make sure that the feature linked to an existing provider are correctly persisted in the right namespaces. This reduce risk of name errors, the builder is just a sugar implementation over the `feature.Bundle`, this make sure that even sub feature use the same global registry.